src/rules/ResourceQuoteStyle.js

```diff
@@ -64,155 +119,193 @@ class ResourceQuoteStyle extends ResourceRule {
 
     /**
      * @private
+     * @param {string} src
+     * @param {string} tar
+     * @param {Resource} resource
+     * @param {string} file
+     * @param {string} locale
      */
-    checkString(src, tar, resource, file, locale, regExps) {
-        quotesAscii.lastIndex = 0;
-        quotesAsciiAlt.lastIndex = 0;
-        regExps.source.quotesNative.lastIndex = 0;
-        regExps.source.quotesNativeAlt.lastIndex = 0;
-        regExps.target.quotesAll.lastIndex = 0;
-        regExps.target.quotesAllAlt.lastIndex = 0;
-        regExps.target.quotesNative.lastIndex = 0;
-        regExps.target.quotesNativeAlt.lastIndex = 0;
```

Comment:
For safety, the lastIndex of all of the regular expressions should be set to 0 so that they are reset for their next use.

Reply:
No need, getRegExps() now creates new instances each time

Reply:
Oh I see